apps/io-func-wallet-solution-backend/src/infra/azure/cosmos/wallet-instance.ts

```diff
@@ -11,6 +13,26 @@ export class CosmosDbWalletInstanceRepository
     this.#container = db.container("wallet-instances");
   }
 
+  get(id: WalletInstance["id"], userId: WalletInstance["userId"]) {
+    return pipe(
+      TE.tryCatch(
+        () => this.#container.item(id, userId).read(),
+        (error) => new Error(`Error getting wallet instance: ${error}`)
+      ),
+      TE.chain(({ resource }) =>
+        pipe(
+          resource,
+          WalletInstance.decode,
+          E.mapLeft(
+            () =>
+              new Error("error getting wallet instance: invalid result format")
+          ),
+          TE.fromEither
```

Comment:
How about throwing an error if the wallet instance is revoked?

Reply:
put it in `getValidWalletInstance` in `src/wallet-instance.ts`